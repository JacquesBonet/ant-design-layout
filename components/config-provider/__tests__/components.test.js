import React from 'react';
import { render } from 'enzyme';
import ConfigProvider from '..';

import { Row, Col } from '../../grid';

describe('ConfigProvider', () => {
  describe('components', () => {
    function testPair(name, renderComponent) {
      // eslint-disable-next-line jest/valid-describe
      describe(`${name}`, () => {
        // normal
        it('normal', () => {
          expect(render(renderComponent({}))).toMatchSnapshot();
        });

        // prefixCls
        it('prefixCls', () => {
          expect(render(renderComponent({ prefixCls: `prefix-${name}` }))).toMatchSnapshot();
        });

        // configProvider
        it('configProvider', () => {
          expect(
            render(
              <ConfigProvider pageHeader={{ ghost: false }} prefixCls="config">
                {renderComponent({})}
              </ConfigProvider>,
            ),
          ).toMatchSnapshot();
        });

        it('configProvider componentSize large', () => {
          expect(
            render(
              <ConfigProvider componentSize="large" prefixCls="config">
                {renderComponent({})}
              </ConfigProvider>,
            ),
          ).toMatchSnapshot();
        });

        it('configProvider componentSize middle', () => {
          expect(
            render(
              <ConfigProvider componentSize="middle" prefixCls="config">
                {renderComponent({})}
              </ConfigProvider>,
            ),
          ).toMatchSnapshot();
        });

        it('configProvider virtual and dropdownMatchSelectWidth', () => {
          expect(
            render(
              <ConfigProvider virtual={false} dropdownMatchSelectWidth={false}>
                {renderComponent({})}
              </ConfigProvider>,
            ),
          ).toMatchSnapshot();
        });
      });
    }

    // Grid
    testPair('Grid', props => {
      const rowProps = {};
      const colProps = {};
      if (props.prefixCls) {
        rowProps.prefixCls = 'prefix-row';
        colProps.prefixCls = 'prefix-col';
      }

      return (
        <Row {...rowProps}>
          <Col {...colProps} span={1} />
        </Row>
      );
    });
  });
});
