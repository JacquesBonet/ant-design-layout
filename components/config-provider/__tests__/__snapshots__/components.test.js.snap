// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigProvider components Grid configProvider 1`] = `
<div
  class="config-row"
>
  <div
    class="config-col config-col-1"
  />
</div>
`;

exports[`ConfigProvider components Grid configProvider componentSize large 1`] = `
<div
  class="config-row"
>
  <div
    class="config-col config-col-1"
  />
</div>
`;

exports[`ConfigProvider components Grid configProvider componentSize middle 1`] = `
<div
  class="config-row"
>
  <div
    class="config-col config-col-1"
  />
</div>
`;

exports[`ConfigProvider components Grid configProvider virtual and dropdownMatchSelectWidth 1`] = `
<div
  class="ant-row"
>
  <div
    class="ant-col ant-col-1"
  />
</div>
`;

exports[`ConfigProvider components Grid normal 1`] = `
<div
  class="ant-row"
>
  <div
    class="ant-col ant-col-1"
  />
</div>
`;

exports[`ConfigProvider components Grid prefixCls 1`] = `
<div
  class="prefix-row"
>
  <div
    class="prefix-col prefix-col-1"
  />
</div>
`;
